### ISSUE TITLE
AvoidMultipleConditionsPerLine: new check

### ISSUE THREAD

**Issue by rnveach** [sevntu-checkstyle/sevntu.checkstyle#564]
Taken from discussion at https://github.com/checkstyle/checkstyle/pull/4148#discussion_r109515706 : 

Code coverage branches should be on their own line.

This will help with code coverage reporting. Reports currently use the same styling as the source code. This is fine until 1 line contains multiple branches. If not all coverage is hit, it makes it harder to determine which branch is failing as it doesn't single out the specific condition.

We should look into either creating this as a new check or see if this is just a specialized configuration of http://checkstyle.sourceforge.net/config_whitespace.html#OperatorWrap (always require new line before `&&`/`||`) .

The following cases should be reported as violations:
````
if (condition1 && condition2) {} // 1 violation
if (condition1 && (condition2 || condition3)) {} // 2 violations
for (int i = 0; condition1 && condition2; i++) {} // 1 violation
while (condition1 && condition2) {} // 1 violation
do {} while (condition1 && condition2) {} // 1 violation
````

The following cases should not be report:
````
if (condition) {}
if (condition & condition2) {} // confirmation that it is seen as single branch because of single '&'
for (int i = 0; condition; i++) {}
while (condition) {}
do (condition) {}

if (condition1
    && condition2) {} // confirmation that it doesn't care where '&&' is on the line
if (condition1
    && condition2
    == value) {}
if (condition1
    && (condition2
        || condition3)) {}
for (int i = 0; condition1 && // confirmation that it doesn't care where '&&' is on the line
    condition2; i++) {}
```

@MEZk @romani Feel free to review and think up any weird situations we should examine and add.

**Comment by MEZk** [comment]
@rnveach 
One more example:

```java
try {
    ...
}
catch (IOException | UnsupportedOperationException | Exception e) { // violation
    ...
}
```
